src/core/mount-setup.c

```diff
@@ -76,10 +76,10 @@ static const MountPoint mount_table[] = {
         { "devpts",      "/dev/pts",                  "devpts",     "mode=620,gid=" STRINGIFY(TTY_GID), MS_NOSUID|MS_NOEXEC,
           NULL,          MNT_IN_CONTAINER           },
 #if ENABLE_SMACK
-        { "tmpfs",       "/run",                      "tmpfs",      "mode=755,smackfsroot=*",  MS_NOSUID|MS_NODEV|MS_STRICTATIME,
+        { "tmpfs",       "/run",                      "tmpfs",      "mode=755,smackfsroot=*",  MS_NOSUID|MS_NOEXEC|MS_NODEV|MS_STRICTATIME,
           mac_smack_use, MNT_FATAL                  },
 #endif
-        { "tmpfs",       "/run",                      "tmpfs",      "mode=755",                MS_NOSUID|MS_NODEV|MS_STRICTATIME,
+        { "tmpfs",       "/run",                      "tmpfs",      "mode=755",                MS_NOSUID|MS_NOEXEC|MS_NODEV|MS_STRICTATIME,
```

Comment:
hmm, we can't really do that i fear. we had discussions about this earlier and there are apps which use /run much like $TMP and drop executable stuff there... While I sympathize with the intention I don't think we can change this nowadays, it's too late. Changing this would be a compat break at very little benefit I figure.

Reply:
In practice that is not as per the runtime dir spec intention, nor what has been allowed on either Debian or Ubuntu since the introduction of `/run`.

Do you know of any software examples that does use executables from /run, for a fact?

Cause if those are generic pieces of software... they would be failing to run on ubuntu/debian since forever. E.g. upstart's mountall as far as I can find the history to hand was mounting `/run` with noexec (but note not nodev) thus I suspect older releases of Fedora/Redhat used to have `/run` as noexec as well. I can certainly dig into the history of that. I think there is a lot of experimental data that both `noexec|nodev` are safe here. I will look into changing `initrmafs-tools` to add nodev to `/run`.

In debian/ubuntu universe this is definately not a new change.

On my system I have:
`
$ sudo find /run -type f -executable
find: ‘/run/user/1000/gvfs’: Permission denied
/run/user/1000/gdm/Xauthority
`
Which is odd, I'm confused why that is an executable given it is data, and cannot be executed...

Reply:
I guess if Ubuntu/Debian had /run with noexec, it should be OK to change it, of course with warnings in NEWS and such.

Reply:
so, thinking back, iirc we went through this before, and there was one issue where some java vm caching thing or so was mmap()ing something in their /var/run with MAP_EXEC where they placed pre-generated code converted from bytecode.  If we drop noexec we also make MAP_EXEC impossible. And this blew up in our then, and is going to blow up again...

Reply:
(also for xdg-runtime-dir at least we explicitly document that the dir must be "fully featured", which probably also includes that you can exec stuff in it. nowawadays XDG_RUNTIME_DIR is separate from /run, but it still sounds surprising if for system stuff we make lesse guarantees than for user-level stuff)

Reply:
Fair enough. I think I still want to do the remount pre-shutdown. Maybe like check NOEXEC flag of /run, and remount without it, if its there. I think that part is genuine and safe.

Reply:
Maybe we could mount `/run` with the noexec flag, but `/run/user/foo` without. It's more important for root anyway.